network/betweenness.go

```diff
@@ -87,3 +93,57 @@ func Betweenness(g graph.Graph) map[int]float64 {
 
 	return cb
 }
+
+// BetweennessWeighted returns the betweenness centrality for nodes in the weighted
+// graph g used to construct the given shortest paths.
+//
+//  C_B(v) = \sum_{s ≠ v ≠ t ∈ V} (\sigma_{st}(v) / \sigma_{st})
+//
+func BetweennessWeighted(g graph.CostGraph, p search.ShortestPaths) map[int]float64 {
+	nodes := g.NodeList()
+	cb := make(map[int]float64, len(nodes))
```

Comment:
I think it is more natural to iterate over nodes from the graph and query the returned map for their centrality (at least, that's what I would do even without knowing this subtle issue). Anyway, it would be better to be consistent and from what you say it seems safer to include them explicitly.